pkg/overlay/overlay.go

```diff
@@ -46,24 +49,82 @@ func MountTemp(store storage.Store, containerId, source, dest string, rootUID, r
 		return mount, "", errors.Wrapf(err, "failed to create the overlay %s directory", workDir)
 	}
 
+	overlayOptions := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s,private", source, upperDir, workDir)
+
+	if unshare.IsRootless() {
+		mountProgram := ""
+
+	loop:
+		for _, i := range store.GraphOptions() {
+			s := strings.SplitN(i, "=", 2)
+			if len(s) != 2 {
+				continue
+			}
+			k := s[0]
+			v := s[1]
+			switch k {
+			case ".mount_program", "overlay.mount_program", "overlay2.mount_program":
+				mountProgram = v
+				break loop
```

Comment:
Why do you need loop: and break loop?  Wouldn't break just work?

Reply:
writing too much C recently.  I assumed it would be part of the switch.